Typehints and debug messages

danielwbn opened the pull request:
No functional changes, just add a few type hints and debug log messages.

In some cases, the check of the result of a wlm library call is delayed until the result has been logged.

Tested on Ubuntu 22.04.5, python 3.10.12, HighFinesse WS8

danielwbn committed:
Add a few helpful type hints

danielwbn committed:
Add a few helpful type hints

danielwbn committed:
Add a few helpful debug messages

danielwbn committed:
Add debug messages

dnadlinger commented on wand/server.py:
```diff
@@ -2,9 +2,13 @@
 from wand.tools import LaserOwnedException, LockException
 
+if TYPE_CHECKING:
+    from frontend.wand_server import WandServer
```
This should at least be `wand.frontend.wand_server`, though really, the better solution would probably be to break the cyclic dependency.

dnadlinger commented:
Sorry, this won't merge cleanly now anymore given the transition to Ruff, but you can just `uv run --with dev poe fmt` on your version and then re-create the commit to follow the formatting changes.

DanielHeinrichToptica committed:
Fix import path for WandServer in type hints

Correct the import statement for WandServer to include the proper
frontend module path for type checking.
